core/modules.rs

```diff
@@ -19,18 +21,142 @@ use std::collections::VecDeque;
 use std::future::Future;
 use std::pin::Pin;
 use std::rc::Rc;
-use std::sync::atomic::AtomicI32;
-use std::sync::atomic::Ordering;
 use std::task::Context;
 use std::task::Poll;
 
-lazy_static::lazy_static! {
-  pub static ref NEXT_LOAD_ID: AtomicI32 = AtomicI32::new(0);
-}
-
 pub type ModuleId = i32;
 pub type ModuleLoadId = i32;
 
+const SUPPORTED_TYPE_ASSERTIONS: &[&str] = &["json"];
+
+/// Throws V8 exception if assertions are invalid
+pub(crate) fn validate_import_assertions(
+  scope: &mut v8::HandleScope,
+  assertions: &HashMap<String, String>,
+) {
+  for (key, value) in assertions {
+    if key == "type" && !SUPPORTED_TYPE_ASSERTIONS.contains(&value.as_str()) {
+      let message = v8::String::new(
+        scope,
+        &format!("\"{}\" is not a valid module type.", value),
```

Comment:
We should start using debug format in place of `\"{}\"`, it adds quotes with necessary escaping
```suggestion
        &format!("{:?} is not a valid module type.", value),
```
Ditto for other `\"{}\"`

Reply:
I don't think we should use debug format for the sake of quoting a string. If we want quotes to strings then it's better to be explicit about that IMO and for example do something like `#""{:?}" is not a valid moulde type."#`. Using debug format is less explicit, requires thinking more about how the value is going to look when using debug format, and can more easily cause issues when the type of the value changes.

For example, here's an example of an issue debug format has caused #13076

Reply:
Good point David, I will revert the change.